TicketsDemo.Domain/DefaultImplementations/PriceCalculationWithCode.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Entities.BookingAggregate;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Domain.DTO;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class PriceCalculationWithCode : IPriceCalculationStrategy
+    {
+        private IRunRepository _runRepository;
+        private ITrainRepository _trainRepository;
+        private IBookingAgencie _bookingAgencies;
+
+        public PriceCalculationWithCode(IRunRepository runRepository, ITrainRepository trainRepository, IBookingAgencie bookingAgencies)
+        {
+            _runRepository = runRepository;
+            _trainRepository = trainRepository;
+            _bookingAgencies = bookingAgencies;
+        }  
+
+        public List<PriceComponent> CalculatePrice(TicketParametersDTO parametrs)
```

Comment:
here we duplicate the logic, described in default price calculation strategy. Instead, please utilize default prica calculation strategy here through the dependency injection